zboss/CMakeLists.txt

```diff
@@ -6,30 +6,35 @@
 
 include(${NRFXLIB_DIR}/common.cmake)
 
+
+# Define a separate interface library for Nordic's extensions to ZBOSS stack
+zephyr_interface_library_named(zboss-extensions)
+
 # Common includes, not related to ZBOSS source code.
-zephyr_include_directories(src/zb_error)
-zephyr_include_directories(include/addons)
+target_include_directories(zboss-extensions INTERFACE
+  src/zb_error
+  include/addons
+)
 
-# Source files
-zephyr_sources(src/zb_error/zb_error_to_string.c)
+# Create a library for Nordic's extensions to ZBOSS stack
+zephyr_library()
 
-# Suppress bitfield compatibility warning
-zephyr_compile_options(-Wno-packed-bitfield-compat)
+# Add source files
+zephyr_library_sources(src/zb_error/zb_error_to_string.c)
 
-# Add compile-time definition, indicating which ZBOSS library will be linked
-if (CONFIG_ZIGBEE_ROLE_END_DEVICE)
-  zephyr_compile_definitions(ZB_ED_ROLE)
-endif()
+# Link with interface libraries - zboss and zboss-extenstions
+zephyr_library_link_libraries(zboss-extensions)
+zephyr_library_link_libraries(zboss)
+
+# Link with Zephyr's default target
+zephyr_link_libraries(zboss-extensions)
```

Comment:
I don't think this is needed.
What you would need, is to `zephyr_library_link_libraries(zboss-extensions)` in the `nrf/subsys/zigbee/` lib.

If the app needs to link to the zboss-extension, then there should be a KConfig entry with:
```
config APP_LINK_WITH_ZBOSS-EXTENSION
    bool "Link application with ZBoss library"
    default y
    help
        Some help text
```

Reply:
I prefer ZBOSS than ZBOSS-EXTENSION to be visible by the user.
Still - this option will be hidden (there is no Zigbee app that does not need ZBOSS lib).